index.js

```diff
@@ -131,16 +131,37 @@ class NetFacility extends Base {
     return seed
   }
 
-  buildFirewall (allowed) {
+  buildFirewall (allowed, allowLocal = false) {
+    // convert keys to Buffer if string
+    allowed = allowed?.map(k => typeof k === 'string' ? Buffer.from(k, 'hex') : k)
+
+    // if firewall enabled, allow from local ip
+    const localIp = allowLocal ? this.getLocalIPAddress() : null
+
     return (remotePublicKey, remoteHandshakePayload) => {
       if (allowed && !libKeys.checkAllowList(allowed, remotePublicKey)) {
+        if (allowLocal && localIp && remoteHandshakePayload?.addresses4) {
+          for (const remoteHost of remoteHandshakePayload.addresses4) {
+            if (remoteHost.host === localIp) return false
+          }
+        }
+
         return true
       }
 
       return false
     }
   }
 
+  getLocalIPAddress () {
+    for (const devices of Object.values(os.networkInterfaces())) {
+      const device = devices.find(d => d.family === 'IPv4' && d.address !== '127.0.0.1' && !d.internal)
+      if (device) return device.address
+    }
+
+    return null
```

Comment:
hmm maybe we should return '127.0.0.1'? not sure what I'm missing here why we ignore '127.0.0.1'

Reply:
client always sends external ip, so internal ip is being filtered out. But it should return localhost when external ip is not found. Method has been updated to return localhost by default.